src/karmabot/commands/control.py

```diff
@@ -0,0 +1,62 @@
+import logging
+from typing import Dict, List
+
+import karmabot.bot as bot
+from karmabot.slack import get_user_id
+
+
+def join_public_channels(**kwargs):
+    """Makes the bot join all public channels he is not in yet"""
+    response = bot.app.client.conversations_list(
+        exclude_archived=True, types="public_channel"
+    )
+    if response["ok"]:
+        public_channels = response["channels"]
+    else:
+        error = "Could not retrieve public channel list"
+        logging.error(error)
+        return error
+
+    channels_joined = []
+    for channel in public_channels:
+        if not channel["is_member"]:  # only join channels you are not already in
+            bot.app.client.conversations_join(channel=channel["id"])
```

Comment:
ah cool so bot auto-joins new channels again? (not sure if that was still working)

Reply:
Yes it should auto-join new channels. This only works for channels that are created while the bot is running.
To join already existing channels DM the bot with `@Karmabot join_public_channels`.

Reply:
Awesome